File: Additional/1047.Remove All Adjacent Duplicates In String.py
class Solution:
    def removeDuplicates(self, S: str) -> str:
        b = list(S)
        curlen = len(b) - 1
        det = True
        for i in range(len(b)-1):      
            if i < curlen:
                if b[i] == b[i+1]:
                    b.pop(i)
                    b.pop(i)
                    curlen -= 2
                    det = False
        if det == True:
            return S
        else:
            c = "".join(b)
            if c == "" or len(c) == 1:
                return c
            else:
                return Solution().removeDuplicates(c)

File: Additional/1047.Remove All Adjacent Duplicates In String.ts
function removeDuplicates(s: string): string {
  let curIdx = 1;
  while (curIdx <= s.length - 1) {
    if (s[curIdx - 1] === s[curIdx]) {
      s = s.substring(0, curIdx - 1) + s.substring(curIdx + 1);
      curIdx -= 1;
    }
    else if (s[curIdx - 1] !== s[curIdx]) {
      curIdx += 1;
    };
  }
  return s;
};
